flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1029,6 +1029,11 @@ private TableResultInternal executeInternal(
                         defaultJobName,
                         jobStatusHookList);
         try {
+            // Current context class loader here is the application class loader.
```

Comment:
@ammu20-dev @davidradl @seb-pereira 
This modifies a pretty critical execution path. How confident are we that it doesn't introduce regressions? Specifically, could any existing functionality—such as built-in serializers for various connector formats—break if libraries are inadvertently loaded from user JARs instead of Flink's parent class loader?

Reply:
@afedulov The expected class loader here is the child class loader FlinkUserCodeClassLoader for resolving dependencies related to external jars. I can see that after https://github.com/apache/flink/commit/64cbcd2d4b360f03f7964edece6f3211d184fb06 the class loader which is invoked here is the FlinkUserCodeClassLoader. Also verified that the changes introduced here is not breaking the CI.

Reply:
@ammu20-dev my concern is that 1.19 is a released version and we need to be extra careful not to break existing functionality when users do a patch version upgrade. As for CI, do you know of a specific test that specifically verifies what I mentioned as my main concern, i.e. that the classes shipped inside of the user JARs for non-shaded libraries that might also happen to be used by the serializers won't get overloaded with a potentially different dependency version and break the binary compatibility? If there no such explicit tests, the green CI might be a false negative.

Reply:
Just to be clear - 2.0 can introduce breaking changes. 1.19.x can't.